tests/test_energy_overflows.py

```diff
@@ -0,0 +1,282 @@
+import jax
+import numpy as np
+import pytest
+import scipy
+from common import GradientTest, prepare_nb_system, prepare_water_system
+
+from timemachine.integrator import FIXED_TO_FLOAT
+from timemachine.potentials import Nonbonded, NonbondedAllPairs, NonbondedPairListNegated
+
+pytestmark = [pytest.mark.memcheck]
+
+
+def verify_energies(a, b):
+    """The energies can be nan, and can't assert nan == nan. This function handles that correctly"""
+    if np.isfinite(a) or np.isfinite(b):
+        assert a == b
+    else:
+        assert not (np.isfinite(a) or np.isfinite(b))
```

Comment:
Could you document intent of this function a bit more?

I think it defines a special case "-inf == +inf == nan", but otherwise uses `a.__eq__`

Reply:
Fixed bug I had introduced in the last commit in [b8e7703](https://github.com/proteneer/timemachine/pull/1084/commits/b8e77037d5478bd1f1cd0f2edf7decd87929d773)